src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -24,46 +28,171 @@ const getInternalLabel = async (event) => {
   return `${lesson.name} -- name`;
 };
 
-// runMetaTestsNeed more specific name later, maybe use "validateInternalTests"
-async function runMetaTests(eventTests, eventMetaTests) {
+const iterateMetaTests = async (eventTests, eventMetaTests, challengeLabel) => {
   const [internalTests, metaTests] = await Promise.all([
     getInternalTests(eventTests),
     getMetaTests(eventMetaTests),
   ]);
+  console.log("<metaTests.length>", metaTests.length);
+  console.log("<metaTests>", metaTests);
+  console.log("<code-challenge>", challengeLabel);
+  console.log("<internalTests>: ", internalTests);
+  for (let i = 0; i < metaTests.length; i++) {
+    runInternalTests(
+      internalTests,
+      metaTests[i].caseCode,
+      challengeLabel,
+      metaTests[i].label,
+      metaTests[i].id,
+      metaTests[i].passes
+    );
+  }
+};
+
+const executeTests = async (
+  metaCaseCode,
+  internalTests,
+  metaLabel,
+  challengeLabel,
+  metaTestId,
+  evalResultShouldBe
+) => {
+  return pMap(
+    internalTests?.filter(removeEmpty) || [],
+    async (internalTest) => {
+      const internalTestLabel = internalTest.label;
+      const internalTestCode = internalTest.internalTestCode;
+      const internalTestId = internalTest.id;
+      // console.log("<internalTest pMap> ", internalTest);
+
+      const newTest = {
+        metaLabel,
+        description: null,
+        userPassed: null,
+        resultType: null,
+        evalResult: null,
+        evalResultShouldBe,
+        internalTestLabel,
+        challengeLabel,
+        metaTestId,
+        internalTestId,
+        metaCaseCode,
+        internalTestCode,
+      };
+      // console.log("newTest name: ", newTest);
+
+      const { userPassed, error, description, resultType, evalResult } =
+        await runTestEvaluator({
+          metaCaseCode,
+          internalTestCode,
+          metaLabel,
+          challengeLabel,
+          metaTestId,
+          internalTest,
+          evalResultShouldBe,
+        });
+      newTest.description = description;
+      newTest.resultType = resultType;
+      newTest.evalResult = evalResult;
+      newTest.userPassed = userPassed;
+
+      if (error) {
+        // @ts-expect-error will fix later
+        const { message, stack } = error;
+        // newTest.message = error.message;
+        // newTest.stack = error.stack;
+        // newTest.userPassed = false;
+        newTest.error = message;
+        newTest.stack = stack;
```

Comment:
It makes more sense to have just one property (error) instead of two (error and stack). Error can be an object that has message and stack as the properties. Otherwise, somebody might not necessarily know that stack is related to error and it leads to bloat of the overall properties